src/JsonSchema/Uri/UriRetriever.php

```diff
@@ -25,6 +25,14 @@
 class UriRetriever implements BaseUriRetrieverInterface
 {
     /**
+     * @var array Map of URL translations
+     */
+    protected $translationMap = array(
+        // use local copies of the spec schemas
+        '|^https?://json-schema.org/draft-(0[34])/schema#?|' => 'package://dist/schema/json-schema-draft-$1.json'
```

Comment:
@shmax The feature itself is generic - it's essentially a regex URI rewriter; you can transform any URI into any other URI.

There are two predefined rules:
 * A rule that allows you to load any schema file from within the package by using a `package://` URL prefix;
 * A rule that rewrites official spec schema URLs into local ones.

The user can add more rules, or not, as they see fit.

Are you able to clarify in more detail what aspect of this you're uncomfortable with? If I can get a better understanding of what you're objecting to, then I can try to come up with an approach you're happier with.